snuba/datasets/errors_replacer.py

```diff
@@ -314,6 +314,55 @@ def process_exclude_groups_and_replacement_types_results(
     return needs_final, exclude_groups, replacement_types
 
 
+def get_latest_replacement_time_by_projects(
+    project_ids: Sequence[int], state_name: Optional[ReplacerState]
+) -> Optional[datetime]:
+    """
+    Given project ids, finds highest timestamps for any FINAL project
+    and group exclude replacemetns and returns returns the highest
+    timestamp found.
+    """
+    p = redis_client.pipeline()
+
+    for project_id in project_ids:
+        exclude_groups_key, _ = get_project_exclude_groups_key_and_type_key(
+            project_id, state_name
+        )
+        project_needs_final_key, _ = get_project_needs_final_key_and_type_key(
+            project_id, state_name
+        )
+        # Gets the item with the highest score for a key, along with the score itself
+        p.zrevrangebyscore(
```

Comment:
In theory you could do `zrevrange` with `0, 0`. Since our data is sorted by timestamp, the value at the end of the set is by definition the value with the highest timestamp.

However I think it would be better to maybe try to move this logic behind the existing redis call.